feat: комплексное тестирование API - snapshot, integration тесты и оптимизация CI

dterenin-the-dev opened the pull request:
## Контекст
В документации сервиса KodySu допущена ошибка в примере ответа на запрос `https://www.kody.su/api/v2.1/search.json`.
## Ожидаемое поведение
Фактический ответ API полностью соответствует документации
## Текущее поведение
Поле, обозначенное в документации как `success`, в ответе API называется `number_success`. Поскольку клиент KodySu был реализован согласно документации, текущее поведение приводит к некорректной десериализации ответов API.
Ниже приведен фактический ответ API:
```json
{
    "success": true,
    "query": "79000000000",
    "quota": 981,
    "numbers": [
    {
        "number_current": "79000000000",
        "number_success": true,
        "number_type_str": "ru_mobile",
        "number_type": 1,
        "def": "960",
        "number": "5256063",
        "code_start": "5140000",
        "code_end": "5259999",
        "operator": "Билайн",
        "operator_full": "ПАО &quot;ВЫМПЕЛКОМ&quot;",
        "region": "Калужская область",
        "time": "3.0",
        "bdpn": false,
        "bdpn_operator": ""
    }
    ]
}
```
## Внесенные изменения
Настройки десериализации JSON ответов от API KodySu были изменены таким образом, чтобы свойству `KodySuResult.Success` соответствовало поле `number_success` в JSON.

dterenin-the-dev committed:
fix: rename api response json property to match actual structure

akrisanov commented:
@dterenin-the-dev спасибо за PR. Посмотри пожалуйста еще на тесты. Возможно их нужно адаптировать, дополнить.

dterenin-the-dev commented:
@akrisanov думаю имеет смысл дополнить `KodySuHttpResponseHandlerTests` тестами на десериализацию raw json, чтобы в дальнейшем избежать непреднамеренных нарушений контракта API с нашей стороны.  Что скажешь?

akrisanov reviewed (changes requested):
Давай попробуем добавить контрактные тесты, подтверждающие необходимость изменений и можно будет вмерживать PR.

akrisanov committed:
feat: оптимизация тестов и внедрение snapshot/integration тестирования

- Добавлено Verify.Xunit для snapshot тестирования API контрактов (13 тестов)
- Реализованы integration тесты с реальным API kody.su (4 теста)
- Оптимизирован CI: параллельные unit/integration тесты с обязательным API ключом
- Исправлено определение типов телефонов (добавлены "Мобильный РФ", "Стационарный РФ")
- Удалено 6 избыточных тестов (75→70), устранено дублирование логики
- Настроена git-интеграция для Verify файлов (.gitignore/.gitattributes)

Результат: комплексное покрытие от unit тестов до реального API контракта

akrisanov commented:
@dterenin-the-dev см. коммит https://github.com/planfact/KodySu.Client/pull/1/commits/7e85e96e45987241c6b65582c62dcddb6ba64373

akrisanov reviewed (approved):
Сделал тестирование более оптимальным, словил расхождения, внес правки в код.
